hw/ip/i2c/data/i2c_testplan.hjson

```diff
@@ -45,170 +49,211 @@
+              - Fill up the rx fifo by sending data bytes in over rx
+              - After a random number(less than filled fifo size) of bytes sent over rx, reset the fifo
 
             Checking:
-              - Ensure that reads to rdata register yield 0s after rx_fifo is reset.'''
+              - Ensure that reads to rdata register yield 0s after rx_fifo is reset
+            '''
       milestone: V2
       tests: [""]
     }
     {
       name: fmt_overflow
-      desc: '''Test fmt_overflow interrupt.
+      desc: '''
+            Test fmt_overflow interrupt.
 
             Stimulus:
-              - Keep writing over 32 bytes of data into wdata register.
+              - Keep sending a number of format byte higher than fmt_fifo depth, delay the target response
 
             Checking:
-              - Ensure excess data bytes are dropped and overflow interrupt is asserted.'''
+              - Ensure excess format bytes are dropped
+              - Ensure intr_fmt_overflow interrupt is asserted
+            '''
       milestone: V2
       tests: [""]
     }
     {
       name: rx_overflow
-      desc: '''Test rx_overflow interrupt.
+      desc: '''
+            Test rx_overflow interrupt.
 
             Stimulus:
-              - Keep sending over 32 bytes of data over rx.
+              - Request a number of byte higher than rx_fifo depth, delay rx_fifo reading
 
             Checking:
-              - Ensure excess data bytes are dropped.'''
+              - Ensure excess data bytes are dropped
+              - Ensure intr_rx_overflow interrupt is asserted
+            '''
       milestone: V2
       tests: [""]
     }
     {
       name: fmt_rx_fifo_full
-      desc: '''Test fmt_fifo and rx_fifo in full states.
+      desc: '''
+            Test fmt_fifo and rx_fifo in full states.
 
             Stimulus:
-              - Send over 32 bytes of data but stop when fmt/rx fifo is full.
+              - Send enough read requests, not read data out,
+                then stop until both fmt and rx fifo is full
 
             Checking:
-              - Check full states by reading fifo status register.'''
+              - Check full states by reading fifo status register
+            '''
       milestone: V2
       tests: [""]
     }
     {
       name: rx_timeout
-      desc: '''Test rx_timeout state.
+      desc: '''
+            Test rx_timeout state.
 
             Stimulus:
-              - Program timeout_ctrl register to randomize the timeout. Send random number of
-                data over rx and read fewer data than sent and let the DUT sit idle or the
-                programmed timeout duration.
+              - Program timeout enable bit into timeout_ctrl register
+              - Program timeout values (higher than host scl clock pulse) into timeout_ctrl register
+              - Configure agent to pull down target(device) scl after the cycle-9th (acknowledge pulse),
```

Comment:
`cycle-9th` do you mean after 9 clock cycles?

Reply:
@cindychip that is true